### ISSUE TITLE
 Adding numerals to the output vocabulary

### ISSUE THREAD

**Issue by maltium** [k2-fsa/icefall#1155]
Usually numbers are normalized in transcripts by transforming them to their equivalent spoken text (24 = "twenty four"). Is there an inherent limitation within RNN-Ts that would prevent us from skipping this step, thereby learning the task of inverse text normalization by itself?

Additionally some words are a concatenation of letters and numbers (for example _The G12 countries_). Assuming that it appears in the training set, one would assume this would be learned to be output this way, is this correct?

**Comment by MicKot** [comment]
This might be problematic: With small numbers it would probably be ok, but the larger you get the more sketchy it gets. Somebody says `1000000000`, do you return 10 tokens or 2: `1` and `000000000`? Do you have examples with somebody saying this number? Probably not. What happens if somebody says `10000000 23`? I think you add more ambiguity and complexity, but you end up doing some kind of ITN anyway.
About `G12` - probably yes, but the more samples of it the better (as with everything). Using Lexicons you can say that `G12` is pronounced however you like and that's normally how you would tackle it.
Hope this answers your question.

**Comment by maltium** [comment]
@MicKot thanks for your insight.

I've been playing with OpenAI's Whisper, and it can handle these cases well. For example it managed to transcribe "1,200,500,225" perfectly. Maybe because it's been trained on so much data it was able to learn these patterns as well as proper text capitalization and punctuation.

I would think that the downside of doing the ITN as a post processing step would be the loss of information. I'm sure that the way words are spoken, for example subtle pauses, would impact the results.

**Comment by MicKot** [comment]
Whisper has a big attentional decoder trained to perform a lot of tasks, and that component gave you your perfect transcription. 
You can incorporate time alignments into ITN for better results. Although in streaming cases alignments are not good.
